sdk/python/feast/diff/FcoDiff.py

```diff
@@ -80,23 +83,138 @@ def tag_proto_objects_for_keep_delete_add(
 FIELDS_TO_IGNORE = {"project"}
 
 
-def diff_between(current: FcoProto, new: FcoProto, object_type: str) -> FcoDiff:
-    assert current.DESCRIPTOR.full_name == new.DESCRIPTOR.full_name
+def diff_registry_objects(current: Fco, new: Fco, object_type: str) -> FcoDiff:
+    current_proto = current.to_proto()
+    new_proto = new.to_proto()
+    assert current_proto.DESCRIPTOR.full_name == new_proto.DESCRIPTOR.full_name
     property_diffs = []
     transition: TransitionType = TransitionType.UNCHANGED
-    if current.spec != new.spec:
-        for _field in current.spec.DESCRIPTOR.fields:
+    if current_proto.spec != new_proto.spec:
+        for _field in current_proto.spec.DESCRIPTOR.fields:
             if _field.name in FIELDS_TO_IGNORE:
                 continue
-            if getattr(current.spec, _field.name) != getattr(new.spec, _field.name):
+            if getattr(current_proto.spec, _field.name) != getattr(
+                new_proto.spec, _field.name
+            ):
                 transition = TransitionType.UPDATE
                 property_diffs.append(
                     PropertyDiff(
                         _field.name,
-                        getattr(current.spec, _field.name),
-                        getattr(new.spec, _field.name),
+                        getattr(current_proto.spec, _field.name),
+                        getattr(new_proto.spec, _field.name),
                     )
                 )
     return FcoDiff(
-        new.spec.name, object_type, current, new, property_diffs, transition,
+        new_proto.spec.name, object_type, current, new, property_diffs, transition,
     )
+
+
+def extract_objects_for_keep_delete_update_add(
+    registry: Registry, current_project: str, desired_repo_contents: RepoContents,
+) -> Tuple[
+    Dict[str, Set[Fco]], Dict[str, Set[Fco]], Dict[str, Set[Fco]], Dict[str, Set[Fco]]
+]:
+    """
+    Returns the objects in the registry that must be modified to achieve the desired repo state.
+
+    Args:
+        registry: The registry storing the current repo state.
+        current_project: The Feast project whose objects should be compared.
+        desired_repo_contents: The desired repo state.
+    """
+    objs_to_keep = {}
+    objs_to_delete = {}
+    objs_to_update = {}
+    objs_to_add = {}
+
+    registry_object_type_to_objects: Dict[str, List[Any]]
+    registry_object_type_to_objects = {
+        "entities": registry.list_entities(project=current_project),
```

Comment:
a less error prone way might be to use the direct class names themselves instead of "entities". Looking at this code, I'd be confused why the other fields in the Registry proto don't exist. 

One way to make this maybe a bit cleaner / less error prone could be to make an enum (https://docs.python.org/3/library/enum.html) so you can always reference the enum as key.